pkg/internal/apis/config/encoding/load.go

```diff
@@ -0,0 +1,104 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package encoding
+
+import (
+	"bytes"
+	"io/ioutil"
+	"os"
+
+	yaml "gopkg.in/yaml.v3"
+
+	"sigs.k8s.io/kind/pkg/apis/config/v1alpha3"
+	"sigs.k8s.io/kind/pkg/errors"
+
+	"sigs.k8s.io/kind/pkg/internal/apis/config"
+)
+
+// Load reads the file at path and attempts to convert into a `kind` Config; the file
+// can be one of the different API versions defined in scheme.
+// If path == "" then the default config is returned
+// If path == "-" then reads from stdin
+func Load(path string) (*config.Cluster, error) {
+	// special case: empty path -> default config
+	// TODO(bentheelder): consider removing this
+	if path == "" {
+		out := &config.Cluster{}
+		config.SetDefaultsCluster(out)
+		return out, nil
+	}
+
+	// load the raw contents
+	raw, err := readAll(path)
+	if err != nil {
+		return nil, err
+	}
+
+	// get kind & apiVersion
+	tm := typeMeta{}
+	if err := yaml.Unmarshal(raw, &tm); err != nil {
```

Comment:
hm, i've heard that v3 enables strict by default for the Unmarshal function.
https://github.com/kubernetes-sigs/yaml/pull/26#issuecomment-510687743

Reply:
it doesn't. we parse a config file in CI.

Reply:
https://github.com/go-yaml/yaml/blob/fc94e3f7165204dd261b8d39d39a1e1773b931c9/yaml.go#L89

Reply:
(I took a pass through the actual yaml.v3 code before switching)

Reply:
i see, perhaps the linked commend was misleading.

Reply:
also the new Node object's `Decode()` does not preserve strict being enabled :disappointed: 

AFAICT implementing the secret "obsolete" (v2 api) custom `YamlUnmarshal` method works fine though.